test/e2e-tests.sh

```diff
@@ -27,8 +27,22 @@ if [ "${EVENTING_KAFKA_BROKER_CHANNEL_AUTH_SCENARIO:-""}" != "" ]; then
   success
 fi
 
+# TODO: going to be reverted once we have the Prow changes in https://github.com/knative/test-infra
 export BROKER_CLASS="Kafka"
```

Comment:
Still hardcoded in Prow jobs. Gonna pass this once https://github.com/knative/test-infra/pull/3466 is merged.

Reply:
```suggestion
# TODO: going to be reverted once we have the Prow changes in https://github.com/knative/test-infra
export BROKER_CLASS=${BROKER_CLASS:-"Kafka"}
```

what about this option? that would facilitate running them locally without specifying the class since it would default to the existing/stable one.

Reply:
I personally always prefer explicit errors over invisible defaulting.

However, I am ok with that if you feel strong about it

Reply:
The job is there, can we fix it?

Reply:
> TODO: going to be reverted once we have the Prow changes in https://github.com/knative/test-infra

This is a bad comment of mine, sorry.

I want to make sure that I am not breaking any E2E tests for the regular KafkaBroker with this PR.
Similarly, in https://github.com/knative-sandbox/eventing-kafka-broker/pull/2459, I am testing if E2E tests are passing for the namespaced broker.

So, I will remove that TODO in a separate PR when I am 100% sure that E2E tests are passing in both class.